src/main/java/seedu/address/logic/parser/RemoveGradeCommandParser.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ASSIGNMENT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_GITHUB;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PATH;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SCORE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SORTORDER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TELEGRAM;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_WEEK;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.RemoveGradeCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new RemoveGradeCommand object
+ */
+public class RemoveGradeCommandParser implements Parser<RemoveGradeCommand> {
+
+    @Override
+    public RemoveGradeCommand parse(String args) throws ParseException {
+        if (args.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveGradeCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_ASSIGNMENT);
+
+        // The third OR condition is to check if there are any other PREFIXES present when
+        // it is not supposed to be.
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ASSIGNMENT)
+                || !argMultimap.getPreamble().isEmpty()
+                || arePrefixesPresent(argMultimap, PREFIX_TELEGRAM, PREFIX_EMAIL, PREFIX_PHONE,
+                PREFIX_GITHUB, PREFIX_SCORE, PREFIX_TAG, PREFIX_WEEK, PREFIX_PATH, PREFIX_SORTORDER)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveGradeCommand.MESSAGE_USAGE));
```

Comment:
Im not sure if that is how the argMultiMap works. I belive is it will only split based on `PREFIX_NAME` and `PREFIX_ASSIGNMENT` and the rest are considered as input for NAME or ASSIGNMENT.